src/body.adoc

```diff
@@ -67,3 +67,636 @@ The Ssplcofi extension guarantees that, for Zihpm counters programmed to count p
+
+==== Machine Count Inhibit Register (`mcountinhibit`)
+
+Bit 1 in `mcountinhibit` controls whether the PDIS counter, `spdiscounter`.COUNT, increments.  When `mcountinhibit`[1]=1, the PDIS counter does not increment.
+
+WARNING: _I'd like to have a way to set `mcountinhibit`[1] on any PDIS LCOFI, or maybe just any LCOFI.  But this probably should be a general mechanism for inhibiting selected counters on LCOFI, and Linux doesn't support any of the existing freeze mechanisms.  This means PDIS will count LCOFI handler instructions, at least until the handler inhibits counters.  But that's the same for all other counters as well.  We could add a (fast-track?) freeze-on-LCOFI extension if we think this is a problem._
+
+==== Supervisor PDIS Control Register (`spdisctl`)
+
+The `spdisctl` register provides supervisor mode access to a subset of `mpdisctl`.
+
+Bit 62 in `spdisctl` is read-only 0. As a result, the M field in `mpdisctl` is not accessible through `spdisctl`. All other `mpdisctl` fields are accessible through `spdisctl`.
+
+For RV32, bits 63:32 of `spdisctl` can be accessed via the `spdisctlh` CSR.
+
+[[pdiscnt]]
+==== Supervisor PDIS Counter Register (`spdiscounter`)
+
+The `spdiscounter` register is a 64-bit WARL register that holds the PDIS counter, as well as the initial counter value used for hardware reload.  For RV32, bits 63:32 of `spdiscounter` can be accessed via the `spdiscounterh` CSR.
+
+.Supervisor PDIS Counter Register (`spdiscounter`)
+[wavedrom, , svg]
+....
+{reg: [
+    {bits:  32, name: 'COUNT'},
+    {bits:  32, name: 'INITVAL'},
+], config:{lanes: 1, hspace:1024}}
+....
+
+[cols="15%,85%",options="header"]
+|====
+| Field | Description
+| COUNT | PDIS counter value.
+| INITVAL | Optional PDIS counter initial value.
+|====
+
+The PDIS counter value, `spdiscounter`.COUNT, is incremented for each decoded instruction that matches the instruction type selected in `mpdisctl`.SEL, while in a privilege mode enabled in `mpdisctl`/`vspdisctl`.  Counting is inhibited when `mcountinhibit`[1]=1.
+
+NOTE: _The PDIS counter is limited to 32 bits since, unlike Zihpm counters, it is used only for sampling, rather than for counting.  When Zihpm counters are used in counting mode they count all events for an application, which can run for minutes or more.  Sampling, on the other hand, requires counters only to count through the sample period, which is typically less than 10ms.  Shorter sample periods (higher sampling rates) are preferred for more accurate and detailed profiles, but minimizing sampling overhead can cause longer periods to be chosen.  With a 32-bit counter, even a 10-wide core running at 6 GHz could accommodate a sample period as long as 71ms._
```

Comment:
If I understood correctly, this counter is counting sampled events. It's a bit strange to refer to the periods in `ms` instead of event counts. But maybe that's just my preference.